tests/queries/0_stateless/01862_dictionary_found_rate.sql

```diff
@@ -0,0 +1,99 @@
+-- TODO: cover other dictionaries
+
+DROP DATABASE IF EXISTS 01862_db;
+CREATE DATABASE 01862_db;
+
+DROP TABLE IF EXISTS 01862_db.simple_key_source_table;
+CREATE TABLE 01862_db.simple_key_source_table
+(
+    id UInt64,
+    value String
+) ENGINE = TinyLog();
+
+INSERT INTO 01862_db.simple_key_source_table VALUES (1, 'First');
+INSERT INTO 01862_db.simple_key_source_table VALUES (1, 'First');
+
+DROP DICTIONARY IF EXISTS 01862_db.simple_key_flat_dictionary;
+CREATE DICTIONARY 01862_db.simple_key_flat_dictionary
+(
+    id UInt64,
+    value String
+)
+PRIMARY KEY id
+SOURCE(CLICKHOUSE(HOST 'localhost' PORT tcpPort() DB '01862_db' TABLE 'simple_key_source_table'))
+LAYOUT(FLAT())
+LIFETIME(MIN 0 MAX 1000);
+
+SELECT * FROM 01862_db.simple_key_flat_dictionary;
+SELECT name, database, found_rate FROM system.dictionaries WHERE database = '01862_db' AND name = 'simple_key_flat_dictionary';
+SELECT * FROM 01862_db.simple_key_flat_dictionary WHERE id = 0 FORMAT Null;
+SELECT name, database, found_rate FROM system.dictionaries WHERE database = '01862_db' AND name = 'simple_key_flat_dictionary';
+SELECT dictGet('01862_db.simple_key_flat_dictionary', 'value', toUInt64(2)) FORMAT Null;
+SELECT name, database, round(found_rate, 2) FROM system.dictionaries WHERE database = '01862_db' AND name = 'simple_key_flat_dictionary';
+
+DROP DICTIONARY 01862_db.simple_key_flat_dictionary;
+
+CREATE DICTIONARY 01862_db.simple_key_hashed_dictionary
+(
+    id UInt64,
+    value String
+)
+PRIMARY KEY id
+SOURCE(CLICKHOUSE(HOST 'localhost' PORT tcpPort() DB '01862_db' TABLE 'simple_key_source_table'))
+LAYOUT(HASHED())
+LIFETIME(MIN 0 MAX 1000);
+
+SELECT * FROM 01862_db.simple_key_hashed_dictionary;
+SELECT name, database, found_rate FROM system.dictionaries WHERE database = '01862_db' AND name = 'simple_key_hashed_dictionary';
+
+DROP DICTIONARY 01862_db.simple_key_hashed_dictionary;
+
+-- FIXME: Cache() is buggy due to another issue
```

Comment:
Will be fixed separately (to avoid mixing bug fixes and new bits)

Reply:
@azat were is bug with Cache ?

Reply:
Incorrect keys metrics, here is the fix - #23929

Reply:
Will remove this after referenced patch will be merged.

Reply:
Merged, need to be removed.